src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs

```diff
@@ -16,31 +16,29 @@ public void AddTokens<TController>(TestServerAction action, TestServerTokenColle
 
             for (var i = 0; i < parameters.Length; i++)
             {
-                if (!IgnoreHeader(parameters[i]))
+                if (parameters[i].ParameterType.IsPrimitiveType()
+                    && !IgnoreHeader(parameters[i]))
                 {
-                    if (IsPrimitiveType(parameters[i].ParameterType))
-                    {
-                        var tokenName = parameters[i].Name.ToLowerInvariant();
-                        var tokenValue = action.ArgumentValues[i].Instance;
+                    var tokenName = parameters[i].Name.ToLowerInvariant();
+                    var tokenValue = action.ArgumentValues.Any(x => x.Key == i) ? action.ArgumentValues[i].Instance : null;
 
-                        if (tokenValue != null)
-                        {
-                            tokens.AddToken(tokenName, tokenValue.ToString(), isConventional: false);
-                        }
-                    }
-                    else if (parameters[i].ParameterType.IsArray
-                       && IsPrimitiveType(parameters[i].ParameterType.GetElementType()))
+                    if (tokenValue != null)
                     {
-                        var arrayValues = (Array)action.ArgumentValues[i].Instance;
+                        tokens.AddToken(tokenName, tokenValue.ToString(), isConventional: false);
+                    }
+                }
+                else if (parameters[i].ParameterType.IsArray
```

Comment:
why do u remove "!IgnoreHeader" in this case?

Reply:
was a mistake. Resolved

Reply:
In preview version, you set one if with ignoreHeader and after the other ifs, I prefer this preview version

Reply:
this version ?